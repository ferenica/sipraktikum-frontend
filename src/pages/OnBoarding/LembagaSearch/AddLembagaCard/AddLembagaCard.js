import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const useStyles = makeStyles((theme) => ({
  card: {
    height: '100%',
    borderRadius: 10,
  },
  actionBotton: {
    margin: '30px 0 20px 0',
  },
  addIcon: {
    fontSize: 35,
    color: 'white',
  },
  title: {
    fontSize: 18,
    color: '#4F4F4F',
  },
  contentCard: {
    'height': '100%',
    'width': '100%',
    'minHeight': '160px',
    'maxWidth': '330px',
    'margin': '10px',
    'display': 'flex',
    'flexDirection': 'column',
    'borderRadius': 10,
    'backgroundColor': '#FF8326',
    'transition': '.15s all ease-in-out',
    '&:hover': {
      transform: 'scale(1.05)',
      backgroundColor: '#FFDEC5',
    },
    'boxShadow': '0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12)',
  },
}));

const AddLembagaCard = () => {
  const classes = useStyles();
  return (
    <Button className={classes.contentCard} href={'/tambah-lembaga'} data-testid="addbtn">
      <div className="button-add" style={{margin: 'auto', textAlign: 'center', color: '#FFFFFF', fontSize: '20px', justifyContent: 'center'}}>
        <AddIcon className={classes.addIcon}></AddIcon>
        Tambah Lembaga
      </div>
    </Button>
  );
};

export default AddLembagaCard;
