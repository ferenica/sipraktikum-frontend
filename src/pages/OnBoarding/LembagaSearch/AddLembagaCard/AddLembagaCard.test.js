import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import {render, fireEvent} from '@testing-library/react';
import {getByTestId} from '@testing-library/dom';
import AddLembagaCard from './AddLembagaCard';
import AddIcon from '@material-ui/icons/Add';

Enzyme.configure({adapter: new EnzymeAdapter()});

describe('LembagaComponent', () => {
  it('should renders correctly', () => {
    const component = shallow(<AddLembagaCard />);
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddIcon></AddIcon>, div);
  });

  it('exist button add lembaga', () => {
    const buttonAdd = shallow(<AddLembagaCard />).find('.button-add');
    expect(buttonAdd.length).toBe(1);
  });

  it('exist button add lembaga', () => {
    const buttonAdd = shallow(<AddLembagaCard />).find('.button-add');
    buttonAdd.simulate('click');
  });

  test('addbtn clicked correctly', () => {
    const dom = render(<AddLembagaCard />);
    const buttonAdd = getByTestId(dom.container, 'addbtn');
    fireEvent.click(buttonAdd);
  });
});
