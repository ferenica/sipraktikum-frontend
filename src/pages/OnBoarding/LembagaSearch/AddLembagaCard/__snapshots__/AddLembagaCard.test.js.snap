// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LembagaComponent should renders correctly 1`] = `ShallowWrapper {}`;
